### PR TITLE
Gursaanj/BFB-195 : SceneSwticher EditorWindow

### PR DESCRIPTION
**Pull request by Gursaanj**
**JIRA Ticket**
https://perigongames.atlassian.net/browse/BFB-191

**Description**
Made a SceneSwitcher Editor Window that will allow users to more quickly go between scenes or load them additively on top of current scenes. Only scenes under the `_BForBoss` subfolder are added into this editor window.

**Screenshots / GIFs**
SceneSwitcher

All scenes under the same parent folder are grouped together. 
The Scene that is highlighted green indicates the current main scene the user is on at the moment.

SceneSwitcherContextMenu

Anytime a user creates, deletes, or renames a scene, the Editor Window, if open, should update automatically. If for some reason, changes have not been updated on the Editor Window,  a manual `Refresh Scenes` button can be used in the context menu.

Favoriting Scenes

A feature I added, when I originally thought we wanted to have all scenes in the Assets folder included for the Editor Window (i.e. the 20+ ECM2 scenes and such) I created a **Favoriting** Feature where user can set their favourite scenes at the top of the window so they don't have to backtrack each time they want to change the scenes they will be working on the most.
***Note** The favourited scenes are saved via `EditorPrefs` so each individual can have their own favourite scenes without influencing any other users.

I definitely see this feature being way more useful in the coming months when our scene count is much higher.

### CODE REVIEW

**Comment by m4rtint** [comment]
just need to fix the title of the PR - @mikeslgoh wondering if you had checks in place to make sure the title is correct.

Also not sure if they files should be within the BForBoss Utilities folder instead of the Root Asset folder

File: Assets/SceneSwitcher/Source/Editor/SceneSwithcerEditorWindow.cs [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+using UnityEngine;
+using UnityEditor;
+using System.IO;
+using PerigonGames;
+using UnityEditor.SceneManagement;
+using UnityEngine.SceneManagement;
+
+public class SceneSwithcerEditorWindow : EditorWindow, IHasCustomMenu
+{
+    private const string PROEJCT_PARENT_FOLDER = "BForBoss/";
```
Comment (unoctium1) [inline Assets/SceneSwitcher/Source/Editor/SceneSwithcerEditorWindow.cs:12]: Just a small typo here

File: Assets/SceneSwitcher/Source/Editor/SceneSwithcerEditorWindow.cs [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+    private void OpenSceneSingleMode(SceneConfigSetup sceneConfig)
+    {
+        EditorSceneManager.OpenScene(sceneConfig.path, OpenSceneMode.Single);
+        _currentSceneConfig = sceneConfig;
+    }
+
+    private void LoadSymbols()
+    {
+        _notFavouriteSymbol = EditorGUIUtility.IconContent("d_Favorite On Icon").image as Texture2D;
+        _favouriteSymbol = EditorGUIUtility.IconContent("d_Favorite Icon").image as Texture2D;
```
Comment (unoctium1) [inline Assets/SceneSwitcher/Source/Editor/SceneSwithcerEditorWindow.cs:248]: Should the strings here be pulled out as consts?

File: Assets/SceneSwitcher/Source/Editor/SceneSwithcerEditorWindow.cs (hunk 1 above)
Comment (unoctium1) [reply Assets/SceneSwitcher/Source/Editor/SceneSwithcerEditorWindow.cs:12]: And on the class name too actually

**Review by unoctium1 — COMMENTED** [review]
This looks great, I really like the favoriting toggle feature

File: Assets/SceneSwitcher/Source/Editor/SceneSwithcerEditorWindow.cs (hunk 1 above)
Comment (Gursaanj) [reply Assets/SceneSwitcher/Source/Editor/SceneSwithcerEditorWindow.cs:12]: haha good eye!

**Review by m4rtint — APPROVED** [review]

**Pull request merged**